linera-execution/src/system.rs

```diff
@@ -362,14 +362,15 @@ impl UserData {
             Some(s) => {
                 // Convert the String to a Vec<u8>
                 let vec = s.into_bytes();
-                if vec.len() == 32 {
-                    // Create an array from the Vec<u8>
-                    let mut array = [0u8; 32];
-                    array.copy_from_slice(&vec);
-                    Some(array)
-                } else {
-                    None
-                }
+
+                // Initialize the array with spaces
+                let mut array = [b' '; 32];
+
+                // Copy bytes from Vec<u8> to the array
+                let len = vec.len().min(32);
+                array[..len].copy_from_slice(&vec[..len]);
```

Comment:
But then that might not be valid Unicode, if byte 32 is not a character boundary. So maybe we do want to check that `vec.len() <= 32`.

Reply:
Thx!
Pls check this update! https://github.com/linera-io/linera-protocol/pull/2369/commits/6d2542773752e8fda443b65027c4d3b1fdbec6aa